miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -73,3 +73,180 @@ export.send_asset.1
     # prepare the stack for return - stack has 6 elements too many
     movupw.3 dropw swap drop swap drop
 end
+
+#! Creates a note without assets.
+#!
+#! Inputs stack: [tag, aux, note_type, RECIPIENT, ...]
+#! Outputs: [note_idx, 0, 0, EMPTY_WORD, ...]
+#!
+#! - tag is the tag to be included in the note.
+#! - aux is the auxiliary data to be included in the note.
+#! - note_type is the note's storage type
+#! - RECIPIENT is the recipient of the note, i.e.,
+#!   hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)
+export.cteate_note
+    # This procedure is written to be executed with `call`. It means that has to be padded to the 
+    # length of 16 to preserve the stack state after the call. 
+    push.0 movdn.7 padw movdnw.2 padw movdnw.3
+    # => [tag, aux, note_type, RECIPIENT, 0, EMPTY_WORD, EMPTY_WORD, ...]
+
+    exec.tx::create_note
+    swap drop # TODO: remove this line after bug in the api.masm `create_note` procedure will be 
+              # fixed
+    # => [note_idx, 0, 0, 0, EMPTY_WORD, EMPTY_WORD, EMPTY_WORD, ...]
+
+    # prepare the stack for return - stack has 9 elements too many
+    movdn.9 dropw dropw drop
+    # => [note_idx, 0, 0, EMPTY_WORD, ...]
+end
+
+#! Add the specified amount of assets to the created note to move them out of the current account to
+#! the specified recipient.
+#!
+#! Inputs stack: [ASSETS_HASH, note_idx, ...]
+#! Advice map: {(ASSETS_HASH, [ASSETS])}
+#! Outputs: [note_idx, EMPTY_WORD, ...]
+#!
+#! - ASSETS_HASH is the sequential hash of the assets which are going to be added to the note.
+#! - [ASSETS] is an array of the assets which are going to be added to the note.
+#!
+#! Panics:
+#! - The fungible asset is not found in the vault.
+#! - The amount of the fungible asset in the vault is less than the amount to be removed.
+#! - The non-fungible asset is not found in the vault.
+export.move_asset_into_note
```

Comment:
I was actually thinking this would be much simpler: it would always take just a single asset (i.e., no need to involve the advice provider). Also, similar to the previous comment, we should transform this procedure to always take 16 elements and return 16 elements. It could look something like this:
```
#! Inputs:  [note_id, ASSET, PAD(11)]
#! Outputs: [ASSET, PAD(12)]
export.move_asset_to_note
```

Reply:
Ah, I got it. So in case we need to add several assets, we will just call `move_asset_into_note` the corresponding number of times. I'll rework it and make a lot more simple.

Reply:
Yes - correct!